test/test_util_flat_map.cpp

```diff
@@ -58,11 +58,57 @@ TEST(Util_FlatMap_Basic)
 
 TEST(Util_FlatMap_Construct)
 {
-    util::FlatMap<std::string, size_t> map({{"foo", 1}, {"bar", 2}, {"ape", 3}});
+    const std::initializer_list<std::pair<std::string, size_t>> init{{"foo", 1}, {"bar", 2}, {"ape", 3}};
+    util::FlatMap<std::string, size_t> map(init);
     CHECK_EQUAL(map.size(), 3);
     CHECK_EQUAL(map.count("foo"), 1);
     CHECK_EQUAL(map.count("bar"), 1);
     CHECK_EQUAL(map.count("ape"), 1);
+
+    std::vector<std::pair<std::string, size_t>> vector{init};
+    util::FlatMap<std::string, size_t> map_from_vector(std::move(vector));
+    CHECK_EQUAL(map_from_vector.size(), 3);
+    CHECK_EQUAL(map_from_vector.count("foo"), 1);
+    CHECK_EQUAL(map_from_vector.count("bar"), 1);
+    CHECK_EQUAL(map_from_vector.count("ape"), 1);
+
+    util::FlatMap<std::string, size_t> map_from_iterators(init.begin(), init.end());
+    CHECK_EQUAL(map_from_iterators.size(), 3);
+    CHECK_EQUAL(map_from_iterators.count("foo"), 1);
+    CHECK_EQUAL(map_from_iterators.count("bar"), 1);
+    CHECK_EQUAL(map_from_iterators.count("ape"), 1);
+}
+
+class HeterogeneousCaseInsensitiveCompare {
+public:
+    using is_transparent = std::true_type;
```

Comment:
this appears to be unused

Reply:
Oh, yeah, I just copied this struct out from HTTPHeaders in http.h where this flatmap change will be used.